src/amo/components/WrongPlatformWarning/index.js

```diff
@@ -75,61 +72,54 @@ export class WrongPlatformWarningBase extends React.Component<InternalProps> {
 
     const newLocation = _correctedLocationForPlatform({
       clientApp,
+      isHomePage,
       location,
       userAgentInfo,
     });
 
-    if (_isFenix(userAgentInfo)) {
+    if (_isFirefoxForIOS(userAgentInfo)) {
+      // Firefox for iOS.
+      message = i18n.gettext(
+        `This add-on is not compatible with this browser. Try installing it on Firefox for desktop.`,
+      );
+    } else if (
+      addon &&
+      _isFirefoxForAndroid(userAgentInfo) &&
+      _isFenixCompatible({ addon })
+    ) {
+      // Compatible with Fenix add-on detail page.
+      message = i18n.sprintf(
+        i18n.gettext(
+          `You can install this add-on in the Add-ons Manager.
+          Learn more about <a href="%(newLocation)s">add-ons for Android</a>.`,
+        ),
+        {
+          newLocation: ANDROID_SUMO_LINK_DESTINATION,
+        },
+      );
+    } else if (newLocation === MOBILE_HOME_PAGE_LINK) {
```

Comment:
mm interesting, shouldn't we have the locale in there?

Reply:
I just tried and browsing AMO locally on Fenix would send me back to the Android homepage with the default locale. I think we can do better (mainly because we did that pretty much everywhere else and we have the current locale easily available)

Reply:
I assumed we were okay with a locale-less URL, but I see your point. It makes the code for detecting that we're going to the mobile home page more complex, but it'll be a better UX.